javascript/src/openai/responses.ts

```diff
@@ -0,0 +1,418 @@
+import OpenAI from "openai";
+import type {
+  LanguageModelInput,
+  ModelResponse,
+  PartialModelResponse,
+  ReasoningPart,
+  ContentDelta,
+  ReasoningPartDelta,
+} from "../schema/index.js";
+import { ContentDeltaAccumulator } from "../utils/stream.utils.js";
+import {
+  convertToOpenAITool,
+  convertToOpenAIToolChoice,
+  convertToOpenAISamplingParams,
+} from "./openai.js";
+import type {
+  OpenAIModelOptions,
+  OpenAIResponsesOptions,
+  ResponseCompletedEventWithUsage,
+} from "./types.js";
+
+/**
+ * OpenAI Responses API client for handling thinking tokens and reasoning
+ */
+export class OpenAIResponsesClient {
+  private openai: OpenAI;
+
+  constructor(private options: OpenAIModelOptions) {
+    this.openai = new OpenAI({
+      baseURL: options.baseURL,
+      apiKey: options.apiKey,
+    });
+  }
+
+  /**
+   * Create a response using the OpenAI Responses API
+   */
+  async createResponse(
+    input: LanguageModelInput,
+    responsesOptions?: OpenAIResponsesOptions,
+  ): Promise<ModelResponse> {
+    const params = this.buildResponsesParams(input, responsesOptions);
+    params.stream = false;
+
+    // Make request to OpenAI Responses API endpoint
+    const response = await this.openai.responses.create(params);
+
+    return this.mapResponseToModelResponse(
+      response as OpenAI.Responses.Response,
```

Comment:
we should not have to make this type assertion here
```suggestion
      response,
```

type assertion usually lead the issue (if it hides an invalid type) Could you remove all type assertion across the code?